oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/OneandoneComputeServiceAdapter.java

```diff
@@ -0,0 +1,217 @@
+   }
+
+   @Override
+   public NodeAndInitialCredentials<Server> createNodeWithGroupEncodedIntoName(String group, String name, Template template) {
+
+      final String dataCenterId = template.getLocation().getId();
+      Hardware hardware = template.getHardware();
+      TemplateOptions options = template.getOptions();
+
+      final String loginUser = isNullOrEmpty(options.getLoginUser()) ? "root" : options.getLoginUser();
+      final String password = options.hasLoginPassword() ? options.getLoginPassword() : Passwords.generate();
+      final String privateKey = options.hasLoginPrivateKey() ? options.getPrivateKey() : null;
+      final org.jclouds.compute.domain.Image image = template.getImage();
+
+      //prepare hdds to provision
+      List<? extends Volume> volumes = hardware.getVolumes();
+      List<Hdd.CreateHdd> hdds = new ArrayList<Hdd.CreateHdd>();
+
+      int i = 1;
+      for (final Volume volume : volumes) {
+         try {
+            Hdd.CreateHdd hdd = Hdd.CreateHdd.create(volume.getSize(), volume.isBootDevice());
+            hdds.add(hdd);
+         } catch (Exception ex) {
+            throw Throwables.propagate(ex);
+
+         }
+      }
+
+      // provision server
+      final Server server;
+      Double cores = ComputeServiceUtils.getCores(hardware);
+
+      try {
+         List<? extends Processor> processors = hardware.getProcessors();
+         org.apache.jclouds.oneandone.rest.domain.Hardware.CreateHardware hardwareRequest
+                 = org.apache.jclouds.oneandone.rest.domain.Hardware.CreateHardware.create(processors.size(), cores, hardware.getRam(), hdds);
+         final Server.CreateServer serverRequest = Server.CreateServer.builder()
+                 .name(name)
+                 .hardware(hardwareRequest)
+                 .rsaKey(options.getPublicKey())
+                 .applianceId(image.getId())
+                 .dataCenterId(dataCenterId)
+                 .powerOn(Boolean.TRUE).build();
+
+         logger.trace("<< provisioning server '%s'", serverRequest);
+
+         server = api.serverApi().create(serverRequest);
+         logger.trace(">> provisioning complete for server. returned id='%s'", server.id());
+
+      } catch (Exception ex) {
+         logger.error(ex, ">> failed to provision server. rollbacking..");
+         throw Throwables.propagate(ex);
+      }
+      LoginCredentials serverCredentials;
+      
+      //if no private key was set use the set/generated password.
+      if (privateKey == null) {
+         serverCredentials = LoginCredentials.builder()
+                 .user(loginUser)
+                 .password(password)
+                 .build();
+      } else {
+         serverCredentials = LoginCredentials.builder()
+                 .user(loginUser)
+                 .privateKey(privateKey)
+                 .build();
+
+      }
+
+      return new NodeAndInitialCredentials<Server>(server, server.id(), serverCredentials);
```

Comment:
In that case, would it make sense to set it to null if the user configured the credentials explicitly, otherwise, read that response and manually set the generated credentials here?